### PR TITLE
issue-1617 Adding wrap definition.

### PR DESCRIPTION
**Pull request by forcepusher**
Fixes https://github.com/AssemblyScript/assemblyscript/issues/1617

- [x] I've read the contributing guidelines

### CODE REVIEW

**Comment by forcepusher** [comment]
That wraps it up.
image

Should I squish the commits ? I've made a bit of a mess right there.

File: src/definitions.ts [hunk 1]
```diff
@@ -496,6 +496,10 @@ export class TSDBuilder extends ExportsWalker {
       sb.push(extendsNode.name.identifier.text); // TODO: fqn?
     }
     sb.push(" {\n");
+    indent(sb, this.indentLevel);
+    sb.push("static wrap(ptr: usize): ");
+    sb.push(name);
+    sb.push("\n");
```
Comment (dcodeIO) [inline src/definitions.ts:502]:
```suggestion
    if (!isInterface) {
      indent(sb, this.indentLevel);
      sb.push("static wrap(ptr: usize): ");
      sb.push(name);
      sb.push("\n");
    }
```

**Review by dcodeIO — COMMENTED** [review]
LGTM, but I think we should not generate the definition for interfaces, as these are merely types currently.

**Comment by forcepusher** [comment]
@dcodeIO Got it. Maybe I should reference https://github.com/AssemblyScript/assemblyscript/blob/master/src/common.ts#L121 as well instead of hardcoding the `usize` into the string ?

File: src/definitions.ts [hunk 2]
```diff
@@ -496,6 +496,12 @@ export class TSDBuilder extends ExportsWalker {
       sb.push(extendsNode.name.identifier.text); // TODO: fqn?
     }
     sb.push(" {\n");
+    if (!isInterface) {
+      indent(sb, this.indentLevel);
+      sb.push("static wrap(ptr: usize): ");
+      sb.push(name);
+      sb.push("\n");
```
Comment (dcodeIO) [inline src/definitions.ts:503]: Looking at this again, is it missing a semicolon here?

File: src/definitions.ts [hunk 3]
```diff
@@ -496,6 +496,13 @@ export class TSDBuilder extends ExportsWalker {
       sb.push(extendsNode.name.identifier.text); // TODO: fqn?
     }
     sb.push(" {\n");
+    if (!isInterface) {
+      indent(sb, this.indentLevel);
+      sb.push("static wrap(ptr: usize): ");
+      sb.push(name);
+      sb.push(";");
+      sb.push("\n");
```
Comment (dcodeIO) [inline src/definitions.ts:504]:
```suggestion
      sb.push(";\n");
```

Comment (forcepusher) [reply src/definitions.ts]: Whoops, was fixing it in a hurry.

File: src/definitions.ts (hunk 2 above)
Comment (forcepusher) [reply src/definitions.ts]: Indeed, missed it.

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**